Core/src/org/sleuthkit/autopsy/datamodel/Md5Node.java

```diff
@@ -53,16 +54,34 @@
     private final String dataSources;
 
     public Md5Node(Md5Metadata data) {
-        super(Children.create(
-                new FileInstanceNodeFactory(data), true),
-                Lookups.singleton(data.getMd5()));
-        
+        super(Children.createLazy(new Md5ChildCallable(data)), Lookups.singleton(data.getMd5()));
         this.commonFileCount = data.size();
         this.dataSources = String.join(", ", data.getDataSources());
         this.md5Hash = data.getMd5();
 
         this.setDisplayName(this.md5Hash);
     }
+    
+    /**
+     * Callable wrapper to further delay lazy ChildFactory creation
+     * and createNodes() call once lazy loading is functional.
+     */
+    private static class Md5ChildCallable implements Callable<Children> {
+        private final Md5Metadata key;
+        private Md5ChildCallable(Md5Metadata key) {
+            this.key = key;
+        }
+        @Override
+        public Children call() throws Exception {
+            //Check that the key has children,
+            //if it doesn't have children, return a leaf:
+            if (key.getMetadata().isEmpty()) {
+                return Children.LEAF;
+            } else {
+                return Children.create(new FileInstanceNodeFactory(key), true);
+            }
+        }
+    }
```

Comment:
My understanding of lazy Children creation is that it is unrelated to lazy child node creation. Lazy Children creation seems to me to be for cases where you want to make different types of Children based on the context at the time of node creation, and it is not something provides a performance gain. Do we have a measurable gain from this? If not, it seems to add complexity for little or no gain.

Reply:
Removed for now.